src/containers/maps/MapVisuals.jsx

```diff
@@ -0,0 +1,171 @@
+      zoom: this.props.mapConfig.map.centre.zoom,
+    });
+    //--------------------------------
+    
+    //Prepare the tile (map base) layers.
+    //--------------------------------
+    const tileLayers = {};
+    this.props.mapConfig.map.tileLayers.map((layer, index) => {
+      const tl = L.tileLayer(layer.url, { attribution: layer.attribution, });
+      tileLayers[layer.name] = tl;
+      if (index === 0) tl.addTo(this.map);  //Use the first tile layer as the default tile layer.
+    });
+    //--------------------------------
+    
+    //Prepare the dynamic data layer.
+    //Starts off empty, but is populated by updateDataLayer().
+    //--------------------------------
+    this.dataLayer = L.geoJson(null, {
+      pointToLayer: this.renderMarker
+    }).addTo(this.map);
+    Actions.getMapMarkers(this.props.mapConfig);
+    //--------------------------------
+    
+    //Prepare additional geographic information layers (park boundaries, etc)
+    //--------------------------------
+    const geomapLayers = {};
+    //--------------------------------
+    
+    //Add standard 'Layer' controls
+    //--------------------------------
+    L.control.layers(tileLayers, {
+      'Data': this.dataLayer,
+      //...geomapLayers
+    }, {
+      position: 'topleft',
+      collapsed: true,
+    }).addTo(this.map);
+    //--------------------------------
+  }
+  
+  //----------------------------------------------------------------
+  
+  updateDataLayer(props = this.props) {
+    if (!this.map || !this.dataLayer || !props.markersData) return;
+    
+    this.dataLayer.clearLayers();
+    this.dataLayer.addData(props.markersData);  //Markers Data must be in GeoJSON format.
+  }
+  
+  renderMarker(feature, latlng) {
+    const count = (feature && feature.properties && feature.properties.count !== undefined)
+      ? feature.properties.count : 0;  //Warning: assumption is `count` is an integer.
+    const radius = 
+      Math.max(Math.min((count - DEFAULT_MARKER.minValue) / DEFAULT_MARKER.maxValue, 1), 0) *
+      (DEFAULT_MARKER.maxRadius - DEFAULT_MARKER.minRadius) +
+      DEFAULT_MARKER.minRadius;
+    
+    const marker = L.circleMarker(latlng, {
+      color: DEFAULT_MARKER.color,
+      fillColor: (count > 0) ? DEFAULT_MARKER.fillColor : DEFAULT_MARKER.emptyFillColor,
+      fillOpacity: DEFAULT_MARKER.fillOpacity,
+      radius,
+    });
+    
+    marker.on('click', this.examineMarker);
+    return marker;
+  }
+  
+  examineMarker(e) {
+    if (!e || !e.target || !e.target.feature || !e.target.feature.properties) return;
+    
+    const cameraId = e.target.feature.properties.id;
+    alert(`TODO: Examine camera ${cameraId}`);  //TODO: Make this a "View a single camera's photos" action.
+  }
+  
+  //----------------------------------------------------------------
+  
+  render() {
+    return (
+      <section ref="mapVisuals" className="map-visuals">
+        <link
```

Comment:
I used the 'component-specific <link>' method instead of adding the CSS link at index.tpl.html because I didn't want the CSS to load when people were only using Intro 2 Astro and not WildCam Darien... but now I realise I may be overthinking it somewhat, since the leaflet.css sits at a mere 14kb uncompressed.

OK, this gets moved to `main.css` with the `-v2` branch. Thanks for the `@import url()` trick; I wasn't aware this was a method of importing external CSS! Sweet!